server/src/main/scala/org/labrad/browser/NodeApi.scala

```diff
@@ -43,6 +46,12 @@ class NodeApi(cxn: LabradConnection)(implicit ec: ExecutionContext) extends Logg
     }
   }
 
+  def getNodeStatus(node: String): Future[NodeStatus] = {
+    cxn.to(node).call("status").map { servers =>
+      NodeStatus(node, getServerStatuses(servers))
+    }
```

Comment:
The only place `getNodeStatus` gets called on the client is [here](https://github.com/labrad/scalabrad-web/pull/329/files#diff-b404fdff99d6a1c62690fd08a74dac31R435). Any exceptions raised on the server will result in a failed scala Future, which then sends the error over json-rpc and results in a failed typescript Promise, which then gets raised in as an exception in our async function when we await that promise. In this particular case, we log an error to the js console. Some day we'll get around to surfacing all these errors in the UI :-)